src/room_entrance.c

```diff
@@ -343,15 +343,14 @@ void generate_creature_for_dungeon(struct Dungeon * dungeon)
         {
             SYNCDBG(8,"The %s will come to player %d even though lair is full",creature_code_name(crmodel),(int)dungeon->owner);
             generate_creature_at_random_entrance(dungeon, crmodel);
-
-            if (dungeon_has_room(dungeon, RoK_LAIR))
+            RoomKind rkind = find_first_roomkind_with_role(RoRoF_LairStorage);
+            if (is_room_of_role_available(dungeon->owner, RoRoF_LairStorage))
```

Comment:
if you have 2 roomkinds with the RoRoF_LairStorage it will only check for the first one,

there's already a dungeon_has_room_of_role function that can be used here

Reply:
Like this?